src/manifest.rs

```diff
@@ -518,3 +699,461 @@ impl From<LocalDependencyManifest> for DependencyManifest {
         Self::Local(value)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    mod raw_manifest_tests {
+        use crate::manifest::{Manifest, RawManifest};
+        use std::str::FromStr;
+
+        #[test]
+        fn test_cloned_manifest_convert_to_exact_same_string() {
+            let manifest = r#"
+            edition = "0.12"
+
+            [package]
+            type = "lib"
+            name = "lib"
+            version = "0.0.1"
+
+            [dependencies]
+            "#;
+
+            let manifest = Manifest::from_str(manifest).expect("should be valid manifest");
+            let cloned_raw_manifest_str = toml::to_string(&RawManifest::from(manifest.clone()))
+                .expect("should be convertable to str");
+            let raw_manifest_str = toml::to_string(&RawManifest::from(manifest))
+                .expect("should be convertable to str");
+
+            assert!(cloned_raw_manifest_str.contains("edition"));
+            assert_eq!(cloned_raw_manifest_str, raw_manifest_str);
+        }
+    }
+    mod manifest_tests {
```

Comment:
```suggestion

    mod manifest_tests {
```

Nit: Missing newline